packages/coffeescript/plugin/compile-coffeescript.js

```diff
@@ -83,8 +83,9 @@ export class CoffeeCompiler extends CachingCompiler {
 
     let sourceMap = JSON.parse(output.v3SourceMap);
 
-    if (source.indexOf('`') !== -1) {
-      // If source contains backticks, pass the coffee output through babel-compiler
+    if (source.match(/(`|import|export|function\*)/)) {
```

Comment:
Also, this regex could be improved (for example, with word boundaries so it doesn't match, say, a comment about `reexportation`) but it might be futile trying to match the CoffeeScript syntax accurately and efficiently.

Perhaps it would be better to just always pass it to `babel-compiler`?  I'm not sure what the original thinking behind the conditional babel compilation was.

At the very least, if this stays there's no need for the outer parenthesis (as there is no capture) which just add extra, potentially-expensive steps for the regex engine.